radix-engine/src/system/system_callback.rs

```diff
@@ -540,7 +540,7 @@ impl<V: SystemCallbackObject> System<V> {
 
     fn validate_intent_hash_uncosted<S: CommitableSubstateStore>(
```

Comment:
My default position is always not to change unless there is a problem. 

I'm not too sure if this will cause any performance issue as the duplicate reads will be from cache.